src/content/plugins.md

```diff
@@ -0,0 +1,325 @@
+
+Let's have a look at the basic building blocks of a Broccoli plugin.
+
+# Broccoli-Plugin base class
+
+```js
+import Plugin from 'broccoli-plugin';
+
+class MyPlugin extends Plugin {
+    constructor (inputNode, options) {
+        super([inputNodes], options);
+    }
+    
+    build() {
+        // A plugin can receive single or multiple inputs nodes/directories
+        // The files are available in this.inputPaths[0]/this.inputPaths[1]...
+        // You can do anything with these files that you can do in node
+        // The output of your plugin must write to this.outputPath
+    }
+}
+```
+
+## Constructor
+
+`inputNode`: This can be one of two things, a string or another Broccoli plugin. If a string is passed, Broccoli
+expects this to be a source directory and automatically converts this into a
+[broccoli-source](https://github.com/broccolijs/broccoli-source).
+
+When constructing your build pipeline you will be passing plugins as inputs to one another, these are commonly
+referred to as `inputNodes`. The `broccoli-plugin` base class expects an array of `inputNodes`, however the 
+majority of plugins will only require a single input, so feel free to pass an array up if only a single input is
+required.
+
+`options`: (all optional)
+
+```js
+{
+    name: '',
+    annotation: '',
+    persistentOutput: false,
+    needsCache: false,
+}
+```
+
+`options.name`: Custom name used when debugging/printing stack traces. Broccoli will use the name of your
+plugin if this field is not supplied.
+
+`options.annotation`: In addition to the plugin name, Broccoli uses annotations to provide a descriptive label used
+during debugging/printing stack traces. This is not often set by the plugin author, but by the consumer of the 
+plugin to tell multiple instances of the same plugin apart.
+
+`options.persistentOutput`: If true, the output directory is not automatically emptied between
+(re)builds. By default, a plugin's `outputPath` is emptied before each (re)build ensuring a 
+consistent output state with every build. If `persistentOutput` is set to true, this cleanup
+will not happen and the output directory will persist. This may be useful if your plugin
+implements caching to allow the build method to be skipped. 
+
+`options.needsCache`: Despite the name, `needsCache` doesn't provide caching for the plugin. If true, a directory
+is created for the plugin to store temporary files that may be needed between rebuilds that are not included in the 
+build output. Broccoli sets the path to this directory to `this.cachePath`.  
+
+## Build
+
+The build method is where the grunt (lol, intentional pun) of the work happens. Build can do anything you can do in
+node. Additionally, if `build()` returns a promise, Broccoli will wait until the promise resolves before continuing
+the rest of the build. Broccoli only builds one plugin at a time in order, from top to bottom of the build graph.
+
+This function will typically access the following only properties:
```

Comment:
The only here is a tad confusing. Maybe:

typically only access the following properties 

or

typically access the following properties